src/validation.cpp

```diff
@@ -5512,12 +5512,10 @@ void ChainstateManager::CheckBlockIndex() const
             //    tip.
             // So if this block is itself better than any m_chain.Tip() and it wasn't in
             // setBlockIndexCandidates, then it must be in m_blocks_unlinked.
-            for (const Chainstate* c : {m_ibd_chainstate.get(), m_snapshot_chainstate.get()}) {
-                if (!c) continue;
-                const bool is_active = c == &ActiveChainstate();
-                if (!CBlockIndexWorkComparator()(pindex, c->m_chain.Tip()) && !c->setBlockIndexCandidates.contains(const_cast<CBlockIndex*>(pindex))) {
+            for (const auto& c : m_chainstates) {
+                if (!CBlockIndexWorkComparator()(pindex, c->m_chain.Tip()) && c->setBlockIndexCandidates.count(const_cast<CBlockIndex*>(pindex)) == 0) {
```

Comment:
nit https://github.com/bitcoin/bitcoin/commit/f85ff75dad4ce8712fe65a636cf36d57b4066785: Any reason to revert contains to count?